text/3318-field-pojection.md

```diff
@@ -610,6 +614,33 @@ impl<T> Arc<T> {
 }
 ```
 
+## Create a field projection operator
+
+In [Gankra's blog post][faultlore] about overhauling raw pointers she talks about using `~` as field
+projection for raw pointers.
+
+We could implement field projection for raw pointers with the current approach, but that would
+result in `x.y`. If we instead adopt Gankra's idea for field projection in general, it would also
+more clearly convey the intent.
+
+### Advantages:
+
+- we can imbue `~` with new meaning, users will not assume anything about what the operator does
+  from other languages
+- it clearly differentiates it from normal field access
+
+### Disadvantages:
+
+- `.` is less confusing to beginners compared to `~`. Other languages use it primarily as a unary
+  binary negation operator
+- `.` can be used on `&mut Struct`, `&Struct` and `Struct`. The first two are outliers, as they do
```

Comment:
the `.` operator has auto-deref, making applying `.` to `&T` and `&mut T` not outliers, that's simply applying deref once. likewise `.` can be applied to `Arc<Vec<T>>` to access methods on `[T]` simply by applying deref twice.

Reply:
you are right, I will fix that.